Frameworks/CoreGraphics/CGContext.mm

```diff
@@ -1646,8 +1647,12 @@ void CGContextFillRects(CGContextRef context, const CGRect* rects, size_t count)
 */
 void CGContextDrawPath(CGContextRef context, CGPathDrawingMode mode) {
     NOISY_RETURN_IF_NULL(context);
-    UNIMPLEMENTED();
-    context->ClearPath();
+    if (context->HasPath()) {
+        ID2D1Geometry* pGeometry;
```

Comment:
Please make this a `ComPtr<ID2D1Geometry>`. Right now you're getting a +1 reference to a Geometry and then dropping it on the floor.

You should almost never use bare COM interface pointers. :)